runners/google-cloud-dataflow-java/src/test/java/org/apache/beam/runners/dataflow/DataflowRunnerTest.java

```diff
@@ -1001,6 +1007,90 @@ public void translate(
     assertTrue(transform.translated);
   }
 
+  @Test
+  public void testMapStateUnsupportedInBatch() throws Exception {
```

Comment:
I went halfway there - I didn't pass in the `StateSpec` because the generics involved might stymie the `DoFnSignature` reflection, but I did factor out the boilerplate pipeline and exception verification.